src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,264 @@
+Within the directory /workspaces/WRoverPlayground/src/topic_training/src, you will find two files called Demo(.cpp or .py). These files will show a basic class containing both a Publisher and Subscriber. There will be comments throughout the files that explain the purpose of specific sections of the code.
+
+Note that there are a few sections of code that are marked as "Optional." You do not need to include everything in the examples when creating your own code. They are simply showing one way of setting up a Publisher and Subscriber in the same file.
+
+One file will be written in C++, and the other will be in Python. It is recommended that you choose the language you are most comfortable with to complete the training. Afterwards, you should discuss with one of the Software Leads to determine which part of the rover you would like to be assigned to. You will be given small tasks first, working your way up to more difficult projects. If these tasks/projects require a different language, then you can look through this training again or take what you know from the first time and learn as you go.
+
+Conintue reading here and go into one of the Demo files based on the language you chose. Make sure you understand what is being said. The general idea of the Demo (.cpp or .py) file is that it takes in data from a separate Publisher file (PlotDemo.cpp or .py) as a float on the topic `plot` and sends that data in the form of a String on the topic `chatter`.
+
+### C++
+
+A basic setup for ROS topics in C++ would look something like this:
+
+``` C++
+ros::init(argc, argv, "Demo");
+ros::NodeHandle nhandle;
+ros::Subscriber sub = nhandle.subscribe("plot", QUEUE_SIZE, chatterCallback);
+ros::Publisher pub = nhandle.advertise<std_msgs::String>("chatter", QUEUE_SIZE);
+```
+
+To break this down,
+
+``` C++
+ros::init(argc, argv, "Demo");
+```
+Initialize ROS and specify the name of the node (which is the name of the file in this case). This requires the parameters from the main function header (argc and argv).
+
+``` C++
+ros::NodeHandle nhandle;
+```
+Create a handle for the node. This will do the actually initialization of the node in the next few lines.
+
+``` C++
+ros::Subscriber sub = nhandle.subscribe("plot", QUEUE_SIZE, chatterCallback);
+```
+Initialize the Subscriber. The subscribe() function is how we tell ROS we want to receive messages on a specific topic. State the name of the topic (`plot`), the queue size (1), and the function that will be called when a message is received (`chatterCallBack(msg)`). The queue size is the maximum number of messages kept as a buffer before throwing away old ones if we are publishing too quickly.
+
+``` C++
+ros::Publisher pub = nhandle.advertise<std_msgs::String>("chatter", QUEUE_SIZE);
+```
+Initialize the Publisher. The advertise() function is how we tell ROS we want to publish on a topic. State the message type we will be publishing (std_msgs::String), the name of the topic (`chatter`), and the queue size (1).
```

Comment:
> The advertise() function is how we tell ROS we want to publish on a topic.

This wording might confuse a new member into thinking that the `advertise()` function publishes a message to the topic when in reality it only creates a publisher object. Might be better to rephrase this.

```suggestion
Initialize the Publisher. The advertise() function creates a publisher object that we can use to publish messages to a topic later on. State the message type we will be publishing (std_msgs::String), the name of the topic (`chatter`), and the queue size (1).
```

Reply:
Fixed in 0be0e6b.